bin/flash.flashair

```diff
@@ -0,0 +1,35 @@
+# Copyright (c) 2015, NVIDIA CORPORATION. All rights reserved.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+# DEALINGS IN THE SOFTWARE.
+
+# It is outside the scope of this script to ensure that the card is online
+UL_DIR=`mktemp -d`
+
+# Run this script to copy files as they must appear on the target into
+# the given directory.
+. "${bin_dir}/${flashair_copy_script}"
+
+if [ ! -z "${flashair_rmlist}" ]; then
+  ARGS="rmlist:${flashair_rmlist}"
+fi
+
+ARGS="${ARGS} push:${UL_DIR}"
+push-flashair.py ${flashair_ip} ${ARGS}
```

Comment:
In my HW setup, I was intending to turn the Pi's power off when testing wasn't running. This will also cut power to the SD cards, since they take power from the Pi's SD slot. The SD cards will take some time to power up and attach to the network when the board is first powered on to run the test. As such, don't we need to retry push-flashair.py until it succeeds here, or would you expect to handle that in the board-power-on script? I.e. to (a) apply Pi reset so the Pi doesn't boot before we're ready, (b) apply Pi power, (c) poll any related resources such as the flashair IP address before completing?

Note to anyone reading this not familiar with flashair: It's an SD form-factor device that acts as an SD card, but also connects to a WiFi network and runs an HTTP server that can access the underlying storage in parallel with accesses over the SD connector.